crypto/mem.c

```diff
@@ -283,19 +124,20 @@ void *CRYPTO_malloc(int num, const char *file, int line)
     if (num <= 0)
         return NULL;
 
-    if (allow_customize)
-        allow_customize = 0;
-    if (malloc_debug_func != NULL) {
-        if (allow_customize_debug)
-            allow_customize_debug = 0;
-        malloc_debug_func(NULL, num, file, line, 0);
+#ifdef CRYPTO_MDEBUG
+    allow_customize = 0;
```

Comment:
There is perhaps an additional issue in that an external malloc implementation would not be able to unset the allow_customize flag and block subsequent attempts to set the memory functions; perhaps CRYPTO_set_mem_functions should do so.